libraries/physics/src/PhysicsEngine.cpp

```
@@ -22,20 +22,43 @@ uint32_t PhysicsEngine::getNumSubsteps() {
     return _numSubsteps;
 }
 
+btHashMap<btHashInt, int16_t> _collisionMasks;
+
+void initCollisionMaskTable() {
+    if (_collisionMasks.size() == 0) {
+        // build table of masks with their group as the key
+        _collisionMasks.insert(btHashInt((int)BULLET_COLLISION_GROUP_DYNAMIC), BULLET_COLLISION_MASK_DYNAMIC);
+        _collisionMasks.insert(btHashInt((int)BULLET_COLLISION_GROUP_STATIC), BULLET_COLLISION_MASK_STATIC);
+        _collisionMasks.insert(btHashInt((int)BULLET_COLLISION_GROUP_KINEMATIC), BULLET_COLLISION_MASK_KINEMATIC);
+        _collisionMasks.insert(btHashInt((int)BULLET_COLLISION_GROUP_MY_AVATAR), BULLET_COLLISION_MASK_MY_AVATAR);
+        _collisionMasks.insert(btHashInt((int)BULLET_COLLISION_GROUP_OTHER_AVATAR), BULLET_COLLISION_MASK_OTHER_AVATAR);
+        _collisionMasks.insert(btHashInt((int)BULLET_COLLISION_GROUP_COLLISIONLESS), BULLET_COLLISION_MASK_COLLISIONLESS);
+    }
+}
+
+// static
+int16_t PhysicsEngine::getCollisionMask(int16_t group) {
+    const int16_t* mask = _collisionMasks.find(btHashInt((int)group));
+    return mask ? *mask : BULLET_COLLISION_MASK_DEFAULT;
+}
+
+QUuid _sessionID;
+
+// static
+void PhysicsEngine::setSessionUUID(const QUuid& sessionID) {
+    _sessionID = sessionID;
+}
+
+// static
+const QUuid& PhysicsEngine::getSessionID() {
+    return _sessionID;
+}
```

This code becomes **static** in the **PhysicsEngine** namespace.

Note: we're currently using a dynamic map between collision groups and their default collision masks.  When I first wrote this code I thought maybe we would dynamically modify this map for special user-defined groups and masks, however now I'm thinking that we won't supply such a feature, so I'll probably replace this map with a **switch** statement soon.